types/vuex-i18n/index.d.ts

```diff
@@ -64,7 +65,7 @@ export interface Ii18n {
   /**
    * get localized string from store in a given language if available.
    */
-  translateIn(locale: string, key: string, options: any, pluralization?: number): string | undefined;
+  translateIn(locale: string, key: string, options?: any, pluralization?: number): string | undefined;
```

Comment:
doesn't look like options is really optional.

```
	if (count > 3) {
	    options = args[3];
	}
…
		} else {
...
		if (count > 2) {
			options = args[2];
			}
```
I would add an extra overload to match the 2nd signature:
>// get localized string from store in a given language if available.
// there are two possible signatures for the function.
// we will check the arguments to make up the options passed.
// 1: locale, key, options, pluralization
// 2: locale, key, defaultValue, options, pluralization

Reply:
I'm certain it's optional. Here's a bit more context...

```javascript
let translateInLanguage = function translateInLanguage(locale) {
	// initialize options
	let options = {};

	// check if a default value was specified and fill options accordingly
	if (count >= 3 && typeof args[2] === 'string') {
		if (count > 3) {
			options = args[3];
		}
	} else {
		if (count > 2) {
			options = args[2];
		}
	}

	// and later...
	return render(/* ... */, /* ... */, options, /* ... */);
};
```

`options` is initialized as an empty object and is only overwritten if it is passed as an argument. It always has a value before being passed to `render`, the only usage.

The `render` function's signature is:
```javascript
let render = function render(locale, translation, replacements = {}, pluralization = null)
```
`options` is passed to `replacements`, which has a default value (apparently, just in case).

It is only used, eventually, by the nested `replace` function, which verifies that requested properties are available before using them.
```javascript
if (replacements[key] !== undefined) {
	return replacements[key];
}
```

However, there is already an overload for the second signature, and I just noticed that I neglected to mark its `options` parameter as optional. I've corrected that discrepancy in a new commit.